### ISSUE TITLE
node.slaveoutput has no key 'metadata' when used with xdist

### ISSUE THREAD

**Issue by jinder1s** [pytest-dev/pytest-metadata#21]
I ran into the following problem while using pytest with xdist. It seems the key 'metadata' sometimes does not exist in node.slaveoutput. My purposed solution is to add a check for key before trying to access it. I am still in middle of looking for other possible solutions, but thought I should put make this issue so that its in your attention. 

Traceback (most recent call last):
  File "/home/jenkins/edx-venv-2.7/edx-venv/local/lib/python2.7/site-packages/_pytest/main.py", line 205, in wrap_session
    session.exitstatus = doit(config, session) or 0
  File "/home/jenkins/edx-venv-2.7/edx-venv/local/lib/python2.7/site-packages/_pytest/main.py", line 249, in _main
    config.hook.pytest_runtestloop(session=session)
  File "/home/jenkins/edx-venv-2.7/edx-venv/local/lib/python2.7/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/home/jenkins/edx-venv-2.7/edx-venv/local/lib/python2.7/site-packages/pluggy/manager.py", line 92, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/home/jenkins/edx-venv-2.7/edx-venv/local/lib/python2.7/site-packages/pluggy/manager.py", line 86, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
  File "/home/jenkins/edx-venv-2.7/edx-venv/local/lib/python2.7/site-packages/pluggy/callers.py", line 203, in _multicall
    gen.send(outcome)
  File "/home/jenkins/edx-venv-2.7/edx-venv/local/lib/python2.7/site-packages/pluggy/callers.py", line 81, in get_result
    _reraise(*ex)  # noqa
  File "/home/jenkins/edx-venv-2.7/edx-venv/local/lib/python2.7/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/home/jenkins/edx-venv-2.7/edx-venv/local/lib/python2.7/site-packages/xdist/dsession.py", line 112, in pytest_runtestloop
    self.loop_once()
  File "/home/jenkins/edx-venv-2.7/edx-venv/local/lib/python2.7/site-packages/xdist/dsession.py", line 135, in loop_once
    call(**kwargs)
  File "/home/jenkins/edx-venv-2.7/edx-venv/local/lib/python2.7/site-packages/xdist/dsession.py", line 170, in worker_workerfinished
    self.config.hook.pytest_testnodedown(node=node, error=None)
  File "/home/jenkins/edx-venv-2.7/edx-venv/local/lib/python2.7/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/home/jenkins/edx-venv-2.7/edx-venv/local/lib/python2.7/site-packages/pluggy/manager.py", line 92, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/home/jenkins/edx-venv-2.7/edx-venv/local/lib/python2.7/site-packages/pluggy/manager.py", line 86, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
  File "/home/jenkins/edx-venv-2.7/edx-venv/local/lib/python2.7/site-packages/pluggy/callers.py", line 208, in _multicall
    return outcome.get_result()
  File "/home/jenkins/edx-venv-2.7/edx-venv/local/lib/python2.7/site-packages/pluggy/callers.py", line 81, in get_result
    _reraise(*ex)  # noqa
  File "/home/jenkins/edx-venv-2.7/edx-venv/local/lib/python2.7/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/home/jenkins/edx-venv-2.7/edx-venv/local/lib/python2.7/site-packages/pytest_metadata/plugin.py", line 104, in pytest_testnodedown
    node.config._metadata.update(node.slaveoutput["metadata"])
KeyError: 'metadata'

**Comment by BeyondEvil** [comment]
Thanks for report things @jinder1s !

Could you share a minimal testcase where this is reproducible?

Please also list what versions of python, pytest and plugins being used.

Thanks!

**Comment by jinder1s** [comment]
Sorry about this, it seems it was due to dependency mismatch between master and worker processes in jenkins. The master had pytest-metadata installed, but the virtualenv in worker process does not. This led to basic assumptions being false. 

This was an issue on my end, but I think others have made the same mistake also. You might want to add a check for metadata as I recommended earlier and give warning to allow others to debug faster. But at the same time, this should not happen if things are set up correctly. 

Thank you for responding so quickly! I will close this issue for now and allow you to decide if this needs to be looked at.
